### PR TITLE
Added logic to prevent button from being active if no records present

### PR DESCRIPTION
**Pull request by FuriousLlama**
# Fixes or Implements [AB#10898](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10898)

-   [x] Enhancement
-   [ ] Bug
-   [ ] Documentation

## Description

If no records are present the button shows as disabled

image

## Testing

-   [ ] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [x] Not Required

### UI Changes

YES 

### Browsers Tested

-   [x] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

-   Fulfills Work Item Requirements

    -   The changes meet/implement story's acceptance criteria. Fixes identified problems if bug.
    -   Changes not directly related to the task requirements need to be documented on the PR.

-   Compilation / Tests

    -   The changes do not break compilation and/or unit or functional tests; unless other item addresses them specifically.

-   Logic Problems / Functional Behaviour

    -   The changes work as intended and do not introduce problems.

-   Performance

    -   The changes do not introduce obvious performance issues.

-   Documented Standards

    -   The changes adhere to the team's documented [Coding Standards](https://github.com/bcgov/healthgateway/wiki/standards).

-   Readability / Maintainability
    -   The changes can be easily understood/read and allow for future enhancements without major refactoring. Readability is preferred over "clever code".
    -   Reasoning for changes needs to be clearly articulated. Disagreements should be arbitrated by a third developer.

### CODE REVIEW

**Review by mk08sh — DISMISSED** [review]
I have a question about the record type selector, if they have no data in a specific type, would they be able to still select that data source?

**Comment by FuriousLlama** [comment]
> I have a question about the record type selector, if they have no data in a specific type, would they be able to still select that data source?

I don't think there is data that can be shown on one type but not in another. Either, there is no data (for any kind) or there is for all since the rendering of it is what we are calling it 'type'

**Review by ninosamson — DISMISSED** [review]

**Review by sslaws — DISMISSED** [review]

File: Apps/WebClient/src/ClientApp/src/components/report/covid19.vue [hunk 1]
```diff
@@ -92,13 +86,29 @@ export default class COVID19ReportComponent extends Vue {
...
+
     private created() {
         this.logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
         this.retrieveLaboratory({ hdid: this.user.hdid }).catch((err) => {
             this.logger.error(`Error loading Covid19 data: ${err}`);
         });
     }
 
+    private mounted() {
+        this.$emit("on-is-empty-changed", this.isEmpty);
```
Comment (tiagograf-AOT) [inline Apps/WebClient/src/ClientApp/src/components/report/covid19.vue:109]: I don't understand the need of this, is it the equivalent to this.onIsEmptyChanged()? if so I'd rather have intelliSense than depending on the method key as a string.

**Review by tiagograf-AOT — DISMISSED** [review]

**Review by MikeLyttle — DISMISSED** [review]

File: Apps/WebClient/src/ClientApp/src/components/report/covid19.vue (hunk 1 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/components/report/covid19.vue]: Good call. Updated it to reference the method instead of a fresh call

**Review by MikeLyttle — APPROVED** [review]

**Review by sslaws — APPROVED** [review]

**Review by mk08sh — APPROVED** [review]

**Pull request merged**